File: backend/app/schemas/provider_profiles.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel

ProviderDeploymentKind = Literal["local", "cloud"]
ProviderValidationStatus = Literal["configured", "validated", "unavailable", "validation_failed"]


class ProviderProfileResponse(BaseModel):
    profile_id: str
    deployment_kind: ProviderDeploymentKind
    display_name: str
    configured: bool
    validation_status: ProviderValidationStatus
    last_validated_at: datetime | None = None
    model_display_name: str | None = None
    service_description: str
    configuration_version: str


class ProviderProfileListResponse(BaseModel):
    items: list[ProviderProfileResponse]


class ProviderValidationStage(BaseModel):
    name: str
    status: Literal["passed", "failed"]
    latency_ms: int
    model_id: str | None = None
    error: str | None = None


class ProviderValidationResponse(BaseModel):
    profile: ProviderProfileResponse
    stages: list[ProviderValidationStage]

File: frontend/src/api/providerProfiles.ts
import { apiRequest } from "./client";

export interface ProviderProfile {
  profile_id: string;
  deployment_kind: "local" | "cloud";
  display_name: string;
  configured: boolean;
  validation_status: "configured" | "validated" | "unavailable" | "validation_failed";
  last_validated_at: string | null;
  model_display_name: string | null;
  service_description: string;
  configuration_version: string;
}

export interface ProviderValidationResponse {
  profile: ProviderProfile;
  stages: Array<{
    name: string;
    status: "passed" | "failed";
    latency_ms: number;
    model_id?: string | null;
    error?: string | null;
  }>;
}

export function listProviderProfiles(signal?: AbortSignal): Promise<{ items: ProviderProfile[] }> {
  return apiRequest<{ items: ProviderProfile[] }>("/ai/provider-profiles", { signal });
}

export function validateProviderProfile(
  profileId: string,
  signal?: AbortSignal,
): Promise<ProviderValidationResponse> {
  return apiRequest<ProviderValidationResponse>(
    `/ai/provider-profiles/${encodeURIComponent(profileId)}/validate`,
    { method: "POST", signal },
  );
}
